### ISSUE TITLE
undefined error while renaming channel

### ISSUE THREAD

**Issue by Bluscream** [TeaSpeak/TeaSpeak#9]
## System Information
***Operating System***: Ubuntu on Windows 10 x64
***TeaSpeak Version***: 1.0.8

## Issue

When i tried to edit the default `TeaSpeak Server` channel i got disconnected with `(undefined error)`

Client Chat:
```
<13:14:50> Successfully renamed channel "[cspacer01]( ͡° ͜ʖ ͡°)" to "[*spacer]( ͡° ͜ʖ ͡°)"
<13:14:59> Successfully renamed channel "[*spacer]( ͡° ͜ʖ ͡°)" to "[*spacer] ( ͡° ͜ʖ ͡°)"
<13:15:18> Disconnected from server (undefined error)
```

Client Log:
```
[OUT] channelgetdescription cid=5 return_code=1:x
[ IN] notifychanneledited cid=5 reasonid=9 channel_description
[ IN] error id=0 msg=ok return_code=1:x
[OUT] channeledit cid=5 channel_name=[*spacer]\s(\s═í┬░\s═£╩û\s═í┬░)
[ IN] notifychanneledited cid=5 channel_name=[*spacer]\s(\s═í┬░\s═£╩û\s═í┬░) invokerid=1 invokername=(\s═í┬░\s═£╩û\s═í┬░) invokeruid=e3dvocUFTE1UWIvtW8qzulnWErI= reasonid=10
[OUT] channelgetdescription cid=5 return_code=1:y
[ IN] notifychanneledited cid=5 reasonid=9 channel_description
[ IN] error id=0 msg=ok return_code=1:y
[OUT] channelgetdescription cid=2 return_code=1:z
[ IN] notifychanneledited cid=2 reasonid=9 channel_description
[ IN] error id=0 msg=ok return_code=1:z
[OUT] channeledit cid=2 channel_name=[*spacer]\s(\s═í┬░\s═£╩û\s═í┬░)
[ IN] notifychanneledited cid=2 channel_name=[*spacer]\s(\s═í┬░\s═£╩û\s═í┬░) invokerid=1 invokername=(\s═í┬░\s═£╩û\s═í┬░) invokeruid=e3dvocUFTE1UWIvtW8qzulnWErI= reasonid=10
```

Server Log:
```
[13:15:15] [DEBUG] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)][Server -> Client][CommandLow] Id '0/0' Command: notifychanneledited cid=2 reasonid=9 channel_description
[13:15:15] [DEBUG] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)][Server -> Client][Command] Id '0/0' Command: error id=0 msg=ok return_code=1:z
[13:15:15] [ERROR]    | Command handling of command channelgetdescription cid='2' return_code='1:z' needs 16ms
[13:15:15] [ERROR]    | Packet handing of Command needs more than 10ms. (17)
[13:15:18] [DEBUG] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)][Client -> Server][Command] Id '57/0' Command: channeledit cid=2 channel_name=[*spacer]\s(\s͡°\s͜ʖ\s͡°)
[13:15:18] [DEBUG] [Permissin] Test Result: Type: 2 Permission: b_client_ignore_antiflood Req Value: 1 Value: 1 Force required: 1 Target: 0x7fac98001350 Result: 1
[13:15:18] [DEBUG] [Permissin] Test Result: Type: 2 Permission: b_channel_modify_name Req Value: 1 Value: 1 Force required: 1 Target: 0x2644bf0 Result: 1
[13:15:18] [DEBUG] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)][Server -> Client][Command] Id '0/0' Command: notifychanneledited cid=2 channel_name=[*spacer]\s(\s͡°\s͜ʖ\s͡°) invokerid=1 invokername=(\s͡°\s͜ʖ\s͡°) invokeruid=e3dvocUFTE1UWIvtW8qzulnWErI= reasonid=10
[13:15:18] [ERROR]    | Command handling of command channeledit cid='2' channel_name='[*spacer] ( ͡° ͜ʖ ͡°)' needs 11ms
[13:15:18] [ERROR]    | Packet handing of Command needs more than 10ms. (12)
[13:15:32] [DEBUG] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)]Client ping timeout
[13:15:32] [INFO] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)] Unregister client connection. Lock: 6/0x7facb8000ae0
[13:15:32] [DEBUG] startoff!
[13:15:32] [INFO] [127.0.0.1:34777/( ͡° ͜ʖ ͡°)] Deleting VoiceClient instance at 0x7facb8000ae0
>
```

**Comment by Bluscream** [comment]
Now after reconnecting i always get `(channel name is already in use)`

Client Chat:
```
<13:15:18> Disconnected from server (undefined error)
<13:19:38> Trying to connect to server on 127.0.0.1
<13:19:38> Welcome
<13:19:38> Welcome on another TeaSpeak server. (Download now and a licence fee is not your cup of tea! www.teaspeak.de)
<13:19:38> Disconnected from server (channel name is already in use)
```

Server Log:
```
[13:19:38] [INFO] [127.0.0.1:34777/aMW|Bluscream]Init done. Having new client
[13:19:38] [INFO] Init done!
[13:19:38] [DEBUG] [127.0.0.1:34777/aMW|Bluscream][Client -> Server][Command] Id '3/0' Command: clientupdate client_badges=overwolf=0:badges=f22c22f1-8e2d-4d99-8de9-f352dc26ac5b,f81ad44d-e931-47d1-a3ef-5fd160217cf8,1cb07348-34a4-4741-b50f-c41e584370f7 return_code=__1_
[13:19:38] [DEBUG] Change server prop virtualserver_client_connections to 7 at server 1
[13:19:38] [DEBUG] [Permissin] Test Result: Type: 2 Permission: b_client_ignore_antiflood Req Value: 1 Value: 1 Force required: 1 Target: 0x264ba70 Result:
[13:19:38] [INFO] [127.0.0.1:34777/aMW|Bluscream]Update client variable via clientupdate. Variable: 'client_badges' value: 'overwolf=0:badges=f22c22f1-8e2d-4d99-8de9-f352dc26ac5b,f81ad44d-e931-47d1-a3ef-5fd160217cf8,1cb07348-34a4-4741-b50f-c41e584370f7'
[13:19:38] [DEBUG] Changed property in db key: client_badges value: client_badges
[13:19:38] [DEBUG] [127.0.0.1:34777/aMW|Bluscream][Server -> Client][Command] Id '0/0' Command: notifyclientupdated clid=1 client_badges=overwolf=0:badges=f22c22f1-8e2d-4d99-8de9-f352dc26ac5b,f81ad44d-e931-47d1-a3ef-5fd160217cf8,1cb07348-34a4-4741-b50f-c41e584370f7
[13:19:38] [DEBUG] [127.0.0.1:34777/aMW|Bluscream][Server -> Client][Command] Id '0/0' Command: error id=0 msg=ok return_code=__1_
[13:19:38] [ERROR]    | Command handling of command clientupdate client_badges='overwolf=0:badges=f22c22f1-8e2d-4d99-8de9-f352dc26ac5b,f81ad44d-e931-47d1-a3ef-5fd160217cf8,1cb07348-34a4-4741-b50f-c41e584370f7' return_code='__1_' needs 35ms
[13:19:38] [ERROR]    | Packet handing of Command needs more than 10ms. (40)
[13:19:38] [DEBUG] [SQL] UPDATE `clients` SET `lastConnect` = :value WHERE `serverId` = :sid AND `cldbid` = :cldbid - 1 - 1515154778 - 2
[13:19:38] [DEBUG] [SQL] UPDATE `clients` SET `connections` = :value WHERE `serverId` = :sid AND `cldbid` = :cldbid - 1 - 7 - 2
[13:19:53] [DEBUG] [127.0.0.1:34777/aMW|Bluscream]Client ping timeout
[13:19:53] [INFO] [127.0.0.1:34777/aMW|Bluscream] Unregister client connection. Lock: 6/0x7facb800c7b0
[13:19:53] [DEBUG] startoff!
[13:19:53] [DEBUG] startoff!
[13:19:53] [DEBUG] startoff!
[13:19:53] [DEBUG] startoff!
[13:19:53] [INFO] [127.0.0.1:34777/aMW|Bluscream] Deleting VoiceClient instance at 0x7facb800c7b0
[13:20:26] [INFO] [WebList] Sending new report.
[13:20:26] [ERROR]    | [WebList] Could not send weblist request. Error: Invalid argument
>
```

**Comment by WolverinDEV** [comment]
Hello,
Thanks for that report I already heard about hat but but I could never reproduce that.
I’m verry happy about your detailed description.
I will try it out later and fix it :)

**Comment by WolverinDEV** [comment]
Hmm,
i cant reproduce that bug.
Maybe can you send me the complete log from <13:14:00> until <13:16:00>?

**Comment by Bluscream** [comment]
Which one?

**Comment by WolverinDEV** [comment]
Server log :)

**Comment by Bluscream** [comment]
[log-2018-01-05_13#003A08#003A44.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607813/log-2018-01-05_13.003A08.003A44.txt)
[log-2018-01-05_13#003A10#003A25.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607814/log-2018-01-05_13.003A10.003A25.txt)
[log-2018-01-05_13#003A22#003A36.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607815/log-2018-01-05_13.003A22.003A36.txt)
[log-2018-01-05_16#003A01#003A25.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607816/log-2018-01-05_16.003A01.003A25.txt)
[log-2018-01-05_16#003A02#003A46.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607817/log-2018-01-05_16.003A02.003A46.txt)
[log-2018-01-05_12#003A11#003A47.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607818/log-2018-01-05_12.003A11.003A47.txt)
[log-2018-01-05_13#003A06#003A55.txt](https://github.com/TeaSpeak/TeaSpeak/files/1607819/log-2018-01-05_13.003A06.003A55.txt)

**Comment by WolverinDEV** [comment]
~ Fixed in 1.0.10-Beta :)
(PS: Sorry for the early close. (Missclicked)

**Comment by Bluscream** [comment]
No problem, thanks for fixing it :)
